### PR TITLE
Add Schema

### PR DESCRIPTION
**Pull request by packapotatoes**
Includes database dump `db.dump` and instructions for import in design/tracker.notes.md.
Added diesel structures in /src/api/types.rs for interacting with database.

### CODE REVIEW

File: src/api/types.rs [hunk 1]
```diff
@@ -1,14 +1,71 @@
...
 }
 
+#[derive(Queryable, Insertable)]
+#[table_name="artifact_names"]
+pub struct ArtifactName {
+	pub name: String,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="version"]
```
Comment (vitiral) [inline src/api/types.rs]: shouldn't this be `versions` or make the other tables singular? Personally I prefer singluar table names (but I don't know the standard).

File: src/api/types.rs [hunk 2]
```diff
@@ -1,14 +1,71 @@
...
+	pub minor: Option<String>,
+	pub patch: Option<String>,
+	pub build: Option<String>,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="test_info"]
+pub struct TestInfo {
+	pub id: i32,
+	pub test_name: String,
```
Comment (vitiral) [inline src/api/types.rs]: so you haven't figured out how to link the test_name, version, etc to the table?

File: src/api/types.rs (hunk 1 above)
Comment (packapotatoes) [inline src/api/types.rs]: After a bit of research, it looks like singular is standard. Will change it.

File: src/api/types.rs (hunk 2 above)
Comment (packapotatoes) [inline src/api/types.rs]: These structs are linked to the table through diesel, using the macros, but the foreign key associations are not in place. I think we need Diesel macros [HasMany](http://docs.diesel.rs/diesel/macro.HasMany.html) and [BelongsTo](http://docs.diesel.rs/diesel/macro.BelongsTo.html). I believe we need those to use JOIN. I'm going to plaly around with it a bit and see if i can figure it out.

File: src/api/types.rs [hunk 3]
```diff
@@ -1,14 +1,71 @@
...
+		major -> Text,
+		minor -> Nullable<Text>,
+		patch -> Nullable<Text>,
+		build -> Nullable<Text>,
+	}
+}
+
 table! {
-	test_name (id) {
+	test_info (id) {
```
Comment (packapotatoes) [inline src/api/types.rs]: I think this table should instead be called `test_run` or something similar. We are not storing data about the test itself, but about the test run.

@vitiral @TheAustinJones @jmcquown

Comment (jmcquown) [inline src/api/types.rs]: Yeah that sounds fine to me

**Pull request merged**